client-js/main/client/abstract-backend-client.js

```diff
@@ -0,0 +1,279 @@
+/*
+ * Copyright 2018, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+"use strict";
+
+import {Type, TypedMessage} from './typed-message';
+import {CommandHandlingError, CommandValidationError, SpineError} from './errors';
+import TypeParsers from './parser/type-parsers';
+import KnownTypes from './known-types';
+import {Status} from '../proto/spine/core/response_pb';
+import {BackendClient} from './backend-client';
+
+/**
+ * A utility which converts the JS object to its Protobuf counterpart.
+ */
+class ObjectToProto {
+
+  /**
+   * Converts the object to the corresponding Protobuf message.
+   *
+   * The input object is supposed to be a Protobuf message representation, i.e. all its attributes should correspond to
+   * the fields of the specified message type.
+   *
+   * @param {Object} object an object to convert
+   * @param {Type} type a type of the corresponding Protobuf message
+   */
+  static convert(object, type) {
+    const typeUrl = type.url().value();
+    const parser = TypeParsers.parserFor(typeUrl);
+    const proto = parser.fromObject(object);
+    return proto;
+  }
+}
+
+/**
+ * Matches the static information about the query with the queried entity type gathered at runtime.
+ */
+class TypedQuery {
+
+  /**
+   * @param {!spine.client.Query} query an underlying proto message representing the query
```

Comment:
Several issues here:
* Don't we have a type URL defined inside the `Query` instance? With it, we could use known types to fetch the `Type` by the type URL.
* According to the class description this object isn't a query per se. I would say that conversion of the response should be done by the routine receiving the raw response which would call `ObjectToProto` utility.